studioGhibliApi.js

```diff
@@ -0,0 +1,103 @@
+let container;
+let select;
+let opt;
+let ghibliArray;
+
+
+document.addEventListener('DOMContentLoaded', () => {
+    getMovieTitle()
+    let form = document.querySelector('form')
+    form.addEventListener('submit', (event) => {
+        submitReview(event);
+    })
+})
+const getApiData = async () => {
+    const movies = await axios.get('https://ghibliapi.herokuapp.com/films')
+    console.log(movies.data);
+    return movies.data
+}
+
+const submitReview = (event) => {
+    event.preventDefault()
+    opt = getSelectedMovie(select).text.bold()
+    let list = document.querySelector('ul')
+    let noInput = document.querySelector('#error')
+    let textInput = document.querySelector('#textField').value
+    let newReview = document.createElement('li')
+    if (textInput === '') {
+        noInput.innerText = `Please enter a valid review`
+        newReview.innerText = ''
+    } else {
+        noInput.innerText = ''
+        newReview.innerHTML = ` ${opt}: ${textInput}`
+        list.appendChild(newReview)
+    }
+    document.querySelector('#textField').value = ''
+}
+
+const pullFromHtml = async () => {
+    return select = document.querySelector('select')
+}
+
+const getMovieTitle = async () => {
+    select = await pullFromHtml();
+    ghibliArray = await getApiData();
+    for (let i = 0; i < ghibliArray.length; i++) {
+        let options = document.createElement('option')
+        options.value = ghibliArray[i].title;
```

Comment:
This function is fetching all the movies and populating the select options with the movie titles. After that this function calls `getMovieInfo()` which sets up the event listener for the select on `change`. This function not only is its name not conveying enough of what it does, it is doing quite a lot. I think `getMovieInfo()` should be called something like `setupSelectListener()` and it shouldn't be called here, it should be called inside the listener for `DOMContentLoaded`.